awsshell/wizard.py

```diff
@@ -1,53 +1,105 @@
+import sys
 import json
 import jmespath
 import botocore.session
 from botocore import xform_name
+from awsshell.resource import index
 
 
 class WizardException(Exception):
+    """Base exception class for the Wizards"""
     pass
 
 
-class Wizard(object):
+class WizardLoader(object):
+    """This class is responsible for searching various paths to locate wizard
+    models. Given a wizard name it will return a wizard object representing the
+    wizard. Delegates to botocore for finding and loading the JSON models.
+    """
+
+    def __init__(self):
+        self._session = botocore.session.get_session()
+        self._loader = self._session.get_component('data_loader')
+
+    def load_wizard(self, name):
+        """Given a wizard's name, returns an instance of that wizard.
 
-    def __init__(self, filename=None):
-        self._env = Environment()
-        if filename:
-            self.load_from_json(filename)
+        :type name: str
+        :param name: The name of the desired wizard.
+        """
+        # TODO possible naming collisions here, always pick first for now
+        # Need to discuss and specify wizard invocation
+        services = self._loader.list_available_services(type_name=name)
```

Comment:
`type_name` would just be `'wizard-1'`. That will only change if there is a major version bump in the spec.